pkg/fetch/github/github_test.go

```diff
@@ -0,0 +1,186 @@
+//go:build !integration
+
+package github
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestParseTree(t *testing.T) {
+	type treeEntry struct {
+		FileMetadata
+		Object struct {
+			FileContents "graphql:\"... on Blob\""
+		}
+	}
+
+	type test struct {
+		name    string
+		entries []struct {
+			FileMetadata
+			Object struct {
+				FileContents "graphql:\"... on Blob\""
+			}
+		}
+		expectedResults   []*FileInfo
+		expectedRemaining []string
+	}
+
+	tests := []test{
+		{
+			name: "empty root dir",
+			entries: []struct {
+				FileMetadata
+				Object struct {
```

Comment:
:wrench: Having to repeat these nested/anonymous struct definitions is tedious and reduces the clarity of the tests.

It would be better to change the definition of the types to use explicitly-named structs and accept the small reduction in legibility there (the query would look less similar to the simple GraphQL equivalent).